jerry-debugger/yarn.lock

```diff
@@ -0,0 +1,3002 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
```

Comment:
What is the purpose of this file? Do we download and run scripts from the internet inside corporate/university firewall?

Reply:
It's yarn's equivalent of `package-lock.json`:
https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/

https://docs.npmjs.com/files/package-lock.json

Reply:
Looks dangerous from security perspective.

Reply:
Same thing here:
https://github.com/Samsung/iotjscode/blob/master/package-lock.json

Reply:
JFYI, if you don't want to download stuff from the interwebs, it's possible to create an ["offline mirror"] as well and check that in.

It's also possible to roll up the sources and its the dependencies into a single blob of JS (executable by Node.js) by using something like rollup or webpack and commit that build output if you want to avoid having to pull in the dependencies from the npm/yarn package registries on the web.

["offline mirror"]: https://yarnpkg.com/blog/2016/11/24/offline-mirror/

Reply:
I am worried about the review time. I need to check each source one-by-one.

Reply:
> I need to check each source one-by-one.

You mean the sources of all the dependencies? Yeah, that will take a while :P

Reply:
Alternative is using sandboxing like twitter:
http://www.nodesummit.com/prior-video/node-summit-2017-twitter-lite-powered-by-node-james-bellenger/

But commiting a full virtual machine seems overkill.

Reply:
Yeah, I agree it's overkill to commit that here (great presentation, btw, thanks for sharing).

I think "trust" is generic and applies to any executable code that you did not write yourself.
People can run it in a VM if they have trust concerns.

I like to limit the number of dependencies we drag in, but at the same time, I'd like to use existing tools (i.e. TypeScript compiler, unit test framework, websocket library, etc.) and not write everything myself.